### DISCUSSION TITLE
ch_PP-OCRv4_rec_hgnet.yml  用这个模型转换成onnx进行识别，速度超级慢，怎么加速？

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#12324] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：
- 版本号/Version：Paddle：  PaddleOCR：2.7.1  问题相关组件/Related components：
- 运行指令/Command Code：
- 完整报错/Complete Error Message：

请尽量不要包含图片在问题中/Please try to not include the image in the issue.
.
ch_PP-OCRv4_rec_hgnet.yml  用这个模型转换成onnx进行识别，速度超级慢，怎么加速？

**Comment by changdazhou** [comment]
使用的什么GPU呢

**Comment by nissansz** [comment]
cpu模型，转onnx也是慢

**Comment by changdazhou** [comment]
建议使用GPU哈，本来ch_PP-OCRv4_rec_hgnet这个模型就比较大，使用CPU会更慢的

**Comment by nissansz** [comment]
发现个问题。
rec模型 用原生gpu python推理，数字都对了，汉字有错误，转换成onnx推理，汉字都对了，数字有错误，不知道是怎么回事

**Comment by nissansz** [comment]
另外，这里说图片经过二值化转换后，再转换为ndarray数组，识别，准确率会高？

https://zhuanlan.zhihu.com/p/639332166

**Comment by changdazhou** [comment]
> 发现个问题。 rec模型 用原生gpu python推理，数字都对了，汉字有错误，转换成onnx推理，汉字都对了，数字有错误，不知道是怎么回事
这个我们记录反馈一下哈

**Comment by changdazhou** [comment]
> 另外，这里说图片经过二值化转换后，再转换为ndarray数组，识别，准确率会高？
> 
> https://zhuanlan.zhihu.com/p/639332166
是有种可能存在的，但不绝对哈，因为可能在某些场景下提前做二值化会过滤掉一部分噪声从而提高识别精度哈

**Comment by nissansz** [comment]
resize放大两倍，有时准确率会提升，有时会降低，怎么处理resize比较好？

**Comment by changdazhou** [comment]
针对不同的场景是会有不同的表现的哈，不建议放大resize，会增加模型的推理时间的哈

**Comment by changdazhou** [comment]
> 发现个问题。 rec模型 用原生gpu python推理，数字都对了，汉字有错误，转换成onnx推理，汉字都对了，数字有错误，不知道是怎么回事

另外，对于这个问题是在同一张图片上进行测试的吗，如果是同一张图片能否提供一下图片我们这边需要测试下哈

**Comment by nissansz** [comment]
测试图片如下。另外有什么好的padding方法吗？
https://github.com/PaddlePaddle/PaddleOCR/issues/5661

![0](https://github.com/PaddlePaddle/PaddleOCR/assets/41010669/6b954805-b960-4bd4-839e-015f8fe2089e)

**Comment by MingsYang** [comment]
ocr = PaddleOCR(
            show_log=False, use_onnx=True,
            cls_model_dir="=data/cls.model.onnx",
            det_model_dir="=data/det.model.onnx",
            rec_model_dir="=data/rec.model.onnx",
            use_angle_cls=True, use_gpu=True)
 cpu推理v4慢，改用gpu没生效，有什么原因吗

> 建议使用GPU哈，本来ch_PP-OCRv4_rec_hgnet这个模型就比较大，使用CPU会更慢的

ocr = PaddleOCR(
            show_log=False, use_onnx=True,
            cls_model_dir="=data/cls.model.onnx",
            det_model_dir="=data/det.model.onnx",
            rec_model_dir="=data/rec.model.onnx",
            use_angle_cls=True, use_gpu=True)
 cpu推理v4慢，改用gpu没生效，有什么原因吗
